### ISSUE TITLE
SignalFX Support

### ISSUE THREAD

**Issue by Disturbing** [argoproj/argo-rollouts#1046]
# Summary

Would love to use SignalFX with Argo Rollouts

# Use Cases

Be able to do metrics / analysis on rollouts

---
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by kostis-codefresh** [comment]
Since version 1.5 Argo Rollouts supports plugins. So anybody can add support for SignalFX outside of the core Argo Rollouts source code https://argo-rollouts.readthedocs.io/en/stable/analysis/plugins/

**Comment by js4683** [comment]
I built an out-of-tree metric plugin for SignalFx / Splunk Observability Cloud, following the RPC metric-provider plugin path discussed in this issue.

Repo: https://github.com/js4683/rollouts-plugin-metric-signalfx
Release: https://github.com/js4683/rollouts-plugin-metric-signalfx/releases/tag/v0.1.0

The plugin implements the standard `RpcMetricProvider` contract (`InitPlugin`, `Run`, `Resume`, `Terminate`, `GarbageCollect`, `Type`, and `GetMetadata`), executes SignalFlow programs through `github.com/signalfx/signalflow-client-go/v2`, and supports `max`, `min`, `avg`, `sum`, `count`, and `latest` aggregation.

The README and `examples/analysis-template.yaml` include installation through `metricProviderPlugins`, Secret-backed `SIGNALFX_ACCESS_TOKEN` authentication, and an AnalysisTemplate example.

Validation for this release includes:
- `go test -race ./...`
- `go vet ./...`
- Linux amd64 and arm64 binaries with SHA-256 checksums
- Process-level RPC contract coverage and local SignalFlow fake-backend tests

Known limitations are documented: the plugin is alpha, synchronous and bounded, creates one client per `Run`, has no transparent SignalFlow retry, and has not yet been validated against a live SignalFx realm. I would welcome feedback and, if useful, a documentation entry in `docs/analysis/plugins.md` or a future donation to `argoproj-labs` following the existing plugin precedent.

**Comment by kostis-codefresh** [comment]
@js4683 Will you continue working on this plugin? Do you want to place the code in argoproj labs like the rest of the plugin and also add an entry in the documentation under https://argo-rollouts.readthedocs.io/en/stable/analysis/plugins/ ?

**Comment by js4683** [comment]
@kostis-codefresh Yes, I will continue working on this plugin. I am open to adding it in the argoproj labs and adding docs. Is it ok if I open a PR against the official code to add it there?

**Comment by kostis-codefresh** [comment]
@js4683 I invited you here https://github.com/argoproj-labs/rollouts-plugin-metric-signalfx

After you add the code there, feel free to open a PR for this page. https://argo-rollouts.readthedocs.io/en/latest/analysis/plugins/

Let me know if you need anything else. Thanks!
